app/controllers/workshop_logs_controller.rb

```diff
@@ -36,30 +30,24 @@ def update
         file.update(report_id: @workshop_log.id, owner_type: @workshop_log.owner_type, owner_id: @workshop_log.owner_id)
         file.save
       end
-      flash[:alert] = 'Thanks for reporting on a new workshop. '
+      flash[:alert] = 'Thanks for reporting on a workshop.'
       redirect_to root_path
     else
-      flash[:alert] = 'ERROR!'
+      set_form_variables
       render :edit
     end
   end
 
   def create
     set_default_values
-    @agencies = current_user.projects.uniq
```

Comment:
not needed unless create fails, so deleted these and added `set_form_variables` in the else clause